lib/ansible/config/data.py

```diff
@@ -17,11 +17,45 @@
 
 # Make coding more python3-ish
 from __future__ import (absolute_import, division, print_function)
+from ansible.module_utils._text import to_text
+from ansible.errors import AnsibleOptionsError
 __metaclass__ = type
 
-from collections import namedtuple
 
-Setting = namedtuple('Setting','name value origin')
+class Setting(object):
```

Comment:
this is a setting 'resolved' not the definition, you are mixing the concepts here.